models/classes/class.QtiTestCompiler.php

```diff
@@ -777,12 +782,44 @@ private function getServiceLocator()
     }
 
     /**
+     * Compile AssessmentItemRef Href Indexes
+     * 
+     * This method indexes the value of $assessmentItemRef->href by $assessmentItemRef->identifier for later
+     * usage at delivery time (for fast access).
+     * 
+     * @param \qtism\data\AssessmentItemRef $assessmentItemRef
+     */
+    protected function compileAssessmentItemRefHrefIndex(AssessmentItemRef $assessmentItemRef)
+    {
+        $compiledDocDir = $this->getPrivateDirectory();
+        
+        $compiledDocDir->write(
+            self::buildHrefIndexPath($assessmentItemRef->getIdentifier()), 
+            $assessmentItemRef->getHref()
+        );
+    }
+    
+    /**
      * Get the list of mime types of files that are accepted to be put
      * into the public compilation directory.
      * 
      * @return array
      */
-    static protected function getPublicMimeTypes() {
+    static protected function getPublicMimeTypes()
+    {
         return self::$publicMimeTypes;
     }
+    
+    /**
+     * Build Href Index Path
+     * 
+     * Builds the Href Index Path from given $identifier.
+     * 
+     * @param string $identifier
+     * @return string
+     */
+    static public function buildHrefIndexPath($identifier)
+    {
+        return TAOQTITEST_COMPILED_HREF_INDEX_FILE_PREFIX . md5($identifier) . TAOQTITEST_COMPILED_HREF_INDEX_FILE_EXTENSION;
```

Comment:
I just asked @bugalot about the index. This solution is the faster, as we request only one item when needed, instead of having a big index to load each time. When building the map we are not accessing every single item index, but only one at the beginning to check if we need to build the fallback.